src/listeners/message.ts

```diff
@@ -83,32 +83,78 @@ client.on("message", async message => {
-            const images: (Discord.BufferResolvable | Stream)[] = []
-            message.attachments.forEach(file => images.push(file.attachment))
-            staffMsg.setTitle("View attachments")
-            dmEmbed.setTitle("Attachments sent")
-            staffBots.send({ embed: staffMsg, files: images })
+        const timeToPass = 300, timeForConfirmToExpire = 20 // in seconds
+        if (typeof author.lastStaffMessage === "undefined" || author.lastStaffMessage - timeToPass > message.createdTimestamp) {
+            const confirmEmbed = new Discord.MessageEmbed()
+                .setColor(neutralColor)
+                .setAuthor(getString("outgoingConfirmation", "global"))
+                .setDescription(message.content)
+                .setFooter(getString("outgoingDisclaimerConfirm", "global"));
+            
+            const msg = (await message.channel.send(confirmEmbed))
+            msg.react("✅").then(() => msg.react("❎"))
+            const collector = msg.createReactionCollector((reaction: Discord.MessageReaction, reacter: Discord.User) => (reaction.emoji.name === "✅" || reaction.emoji.name === "❎") && reacter.id === message.author.id, { time: timeForConfirmToExpire * 1000 })
+
+            collector.on("collect", async reaction => {
+                if (reaction.emoji.name === "❎") return msg.delete()
+                db.collection("users").updateOne({ id: message.author.id }, { $set: { lastStaffMessage: message.createdTimestamp } })
+                const staffMsg = new Discord.MessageEmbed()
+                    .setColor(neutralColor)
+                    .setAuthor("Incoming message from " + message.author.tag)
+                    .setDescription(message.content)
+                    .addField("To reply", `\`+dm ${message.author.id} \``)
+
+                const dmEmbed = new Discord.MessageEmbed()
+                    .setColor(successColor)
+                    .setAuthor(getString("outgoing", "global"))
+                    .setDescription(message.content)
+                    .setFooter(getString("outgoingDisclaimer", "global"))
+                if (message.attachments.size > 1) {
+                    const images: (Discord.BufferResolvable | Stream)[] = []
+                    message.attachments.forEach(file => images.push(file.attachment))
+                    staffMsg.setTitle("View attachments")
+                    dmEmbed.setTitle("Attachments sent")
+                    staffBots.send({ embed: staffMsg, files: images })
+                    return message.channel.send(dmEmbed)
+                } else if (message.attachments.size > 0) {
+                    staffMsg
+                        .setTitle("View attachment")
+                        .setImage(message.attachments.first()!.url)
+                    dmEmbed.setTitle("Attachment sent")
+                    staffBots.send(staffMsg)
+                } else staffBots.send(staffMsg) //staff-bots
+                msg.delete()
```

Comment:
Actually at first I made it work like that buy because the message is in DM channels the bot doesn't have permissions to remove the users reactions.
Unless you want me to leave the users reaction.

Reply:
Yeah I think just removing the bot's reactions would be good enough